Sources/StreamChat/WebSocketClient/WebSocketClient.swift

```diff
@@ -65,12 +65,13 @@ class WebSocketClient {
         return pingController
     }()
 
-    private func createEngineIfNeeded(for connectEndpoint: Endpoint<EmptyResponse>) -> WebSocketEngine {
+    private func createEngineIfNeeded(for connectEndpoint: Endpoint<EmptyResponse>) throws -> WebSocketEngine {
```

Comment:
`throws` probably isn't needed as you are handling the creation of the engine.

It may be better to actually keep the `throws` but remove all the internal `do...catch` and just allow the caller to handle it

Reply:
The reason this is throwing is that I want to make it clear, when using this function, that we need to handle it if it errors. Otherwise, it will be easy just to do `engine = createEngineIfNeeded()`. This will be problematic because when we use this, we want to make sure we don't proceed with the logic afterwards.

The reason I'm catching the error is that I want to log it if this fails. But indeed, I can just rethrow the error from the catch statement, instead of wrapping it in `ClientError`

Reply:
> I can just rethrow the error from the catch statement, instead of wrapping it in ClientError

That sounds good :)